returnn/datasets/basic.py

```diff
@@ -951,8 +954,9 @@ def iterate_seqs(self, recurrent_net=True, used_data_keys=None):
           if chunk_step[key] == chunk_step[default_key]:
             raise Exception("Chunking with multiple data-keys of different length: %r" % length)
           else:
-            limit = limit_default = 1
-            if self.min_chunk_size == chunk_size[default_key]:
+            limit = self.min_chunk_size[key] or 1
+            limit_default = self.min_chunk_size[default_key] or 1
+            if self.min_chunk_size[default_key] == chunk_size[default_key]:
```

Comment:
Hm, this code seems somehow wrong to me...
But not just due to your change, but already before.
Or does it make sense for you?
I also mean specifically the two lines below:
```python
limit = chunk_size[key]
limit_default = chunk_size[default_key]
```

I see also a somewhat strange edit history for this.
I very roughly remember that I did this for Tobias, but not sure anymore. Maybe also @vieting.

Who exactly uses `min_chunk_size`? I think in all my setups, it's 0 anyway.

Reply:
If we remove this `if` including its code, is it correct then?

Reply:
This only makes sense to me, if indeed the data-keys had different length and then 
`min_chunk_size == chunk_step` for the default key implies "always full chunks for all keys"

I have kept this only for preserving the old behavior and existing setups. We can indeed get the same behavior with my proposed change if the `min_chunk_size` is set to `chunk_step` for each data-key.

I would be fine with removing this if and its code.

Reply:
Before your change, `min_chunk_size` was anyway for all keys.

After your change, you also fixed the `limit` above. So in any case the logic here is different now. But it looks more correct.

I think we should remove this `if` and its code.

Anyway, this whole code here is only to perform the sanity check, for nothing else.

Reply:
ok, it is removed now.